File: 461.汉明距离.py
#
# @lc app=leetcode.cn id=461 lang=python3
#
# [461] 汉明距离
#

# @lc code=start
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        temp, res = x ^ y, 0
        while temp > 0:
            res += temp & 1
            temp >>= 1
        return res
# @lc code=end

File: 461.汉明距离.go
/*
 * @lc app=leetcode.cn id=461 lang=golang
 *
 * [461] 汉明距离
 */

// @lc code=start
func hammingDistance(x int, y int) (res int) {
	for s := x ^ y; s > 0; s >>= 1 {
		res += s & 1
	}
	return
}

// @lc code=end

